pkg/plugins/autodiscovery/dockercompose/utils_test.go

```diff
@@ -0,0 +1,44 @@
+package dockercompose
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSearchFiles(t *testing.T) {
+
+	gotFiles, err := searchDockerComposeFiles(
+		"testdata/", DefaultFileMatch[:])
+	if err != nil {
+		t.Errorf("%s\n", err)
+	}
+	expectedFiles := []string{
+		"testdata/docker-compose.yaml",
+	}
+
+	assert.Equal(t, expectedFiles, gotFiles)
+}
```

Comment:
```suggestion
func TestSearchFiles(t *testing.T) {
	testdata := []struct {
		name          string
		rootDir       string
		filePatterns  []string
		expectedFiles []string
		expectedErr   error
	}{
		{
			name:         "Nominal case with test data and default file pattern set",
			rootDir:      "testdata/",
			filePatterns: DefaultFileMatch,
			expectedFiles: []string{
				"testdata/docker-compose.yaml",
			},
		},
	}

	for _, tt := range testdata {
		t.Run(tt.name, func(t *testing.T) {
			gotFiles, err := searchDockerComposeFiles(tt.rootDir, tt.filePatterns)

			if tt.expectedErr != nil {
				require.Error(t, err)
				return
			}

			require.NoError(t, err)
			assert.Equal(t, tt.expectedFiles, gotFiles)
		})
	}
}
```

Using iteration on test table data makes it easier for someone else reading/debugging (otherwise it's hard to understand and extend the tests